llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1609,3 +1609,10 @@ bool RISCVTTIImpl::isLSRCostLess(const TargetTransformInfo::LSRCost &C1,
                   C2.NumIVMuls, C2.NumBaseAdds,
                   C2.ScaleCost, C2.ImmCost, C2.SetupCost);
 }
+
+bool RISCVTTIImpl::isLegalMaskedCompressStore(Type *DataTy) {
```

Comment:
This interface needs an alignment parameter doesn't it?

Reply:
For RVV it's not required. The store of the compressed data is unaligned, i.e. it should be possible to use `vse8` for non-naturally aligned memory

Reply:
We would need to scale the VL from the vcpop to use `vse8`.

Doesn't look like alignment was accounted for in the intrinsic definition. masked.gather/scatter had a separate argument. vp.gather/scatter use a pointer attribute. I think we should use the pointer attribute for compress store.

It looks like SelectionDAGBuilder::visitMaskedStore currently sets the alignment to the ABI alignment of the vector type which doesn't make sense.

Reply:
Once we fix the alignment for the intrinsic we'll also need to teach ScalarMaskedMemIntrinsic to preserve it when scalarizing.

Reply:
> Once we fix the alignment for the intrinsic we'll also need to teach ScalarMaskedMemIntrinsic to preserve it when scalarizing.

To emphasize, this should be done in a separate review and should probably be addressed before we continue too far with this one.